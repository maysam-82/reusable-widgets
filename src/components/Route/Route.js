import { useEffect, useState } from 'react';

const Route = ({ path, children }) => {
	const [currentPath, setCuurentPath] = useState(window.location.pathname);
	useEffect(() => {
		const onWindowLocationChanged = () => {
			setCuurentPath(window.location.pathname);
		};
		window.addEventListener('popstate', onWindowLocationChanged);
		return () => {
			window.removeEventListener('popstate', onWindowLocationChanged);
		};
	}, []);
	return currentPath === path ? children : null;
};

export default Route;
