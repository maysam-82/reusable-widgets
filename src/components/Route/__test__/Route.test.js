import React from 'react';
import { shallow, mount } from 'enzyme';
import Route from '../Route';

describe('Route component', () => {
	const ChildComponent = () => (
		<div className="test-component">
			<div>test component</div>
		</div>
	);
	const componentWithTestRoute = shallow(
		<Route path="/test">
			<ChildComponent />
		</Route>
	);

	it('should render Route component', () => {
		expect(componentWithTestRoute).toMatchSnapshot();
	});
	it('should render `Accordion` component as its children', () => {
		expect(componentWithTestRoute.find(ChildComponent)).toBeTruthy();
	});
	describe('When route is not equal to `pathname`', () => {
		const componentWithDifferentRoute = shallow(
			<Route path="/">
				<ChildComponent />
			</Route>
		);
		it('should not return component', () => {
			expect(
				componentWithDifferentRoute.find(ChildComponent).exists()
			).toBeFalsy();
		});
	});
});
