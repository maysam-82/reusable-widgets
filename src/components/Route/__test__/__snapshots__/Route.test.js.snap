// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route component should render Route component 1`] = `<ChildComponent />`;
